LSPS2/README.md

```diff
@@ -0,0 +1,736 @@
+* SHOULD avoid characters outside of the printable ASCII range.
+* SHOULD avoid characters that would require escapes in JSON format.
+
+> **Rationale** The method to generate `promise` is left vague in order
+> to allow LSP implementors the flexibility to structure their code
+> as necessary.
+> The above specification suggests the use of any MAC, which would cover
+> the above requirements, and requires a symmetric key known only by
+> the LSP.
+> Alternately, an LSP might be componentized and use private-public
+> key cryptography, with a signature as the `promise`, so that the
+> component that generates `opening_fee_params` objects is the only
+> one with access to a specific private key, while the rest of the LSP,
+> including the `lsps2.buy` component, only knows the public key.
+> Another alternative for a componentized LSP would be to have the
+> `opening_fee_params` generator know the public key of the
+> `opening_fee_params` validator; the generator creates an ephemeral
+> keypair and uses the ephemeral privkey with the validator pubkey
+> to create a session key for a MAC, then concatenate the ephemeral
+> pubkey with the MAC as the `promise`; the validator takes its
+> private key and uses it with the ephemeral pubkey to regenerate
+> the same session key and validates the MAC.
+
+Clients, when reading the `promise` feild:
+
+* MUST abort this flow if `promise` exceeds 512 bytes (not including the
+  `"` delimiters) when encoded as a UTF8-encoded JSON string.
+* MUST NOT parse or interpret this string, and only provide it
+  verbatim later.
+
+> **Rationale** By using a promise that must be remembered by the
+> client, the LSP does not have to remember what it offered via this
+> API, reducing state storage, especially if the client then decides
+> that the resulting `opening_fee` is too high and ultimately decides
+> not to continue this flow.
+> By using a cryptographically-verifiable method, the LSP can ensure
+> that it will only honour `opening_fee_params` it actually offered.
+> A size limit is imposed so that LSPs cannot burden clients with
+> unreasonable storage requirements.
+
+LSPs, and future versions of this specfication, MUST NOT add fields
+to the `opening_fee_params` object other than the fields listed
+above.
+
+> **Rationale** This step of the flow is how the client learns of
+> the API versions the LSP supports, and the client may support
+> older versions than the LSP does.
+> The calculation of the fee based on `opening_fee_params` needs to
+> be stable so that older clients can still talk to later LSP
+> versions.
+>
+> If a future version of this spec absolutely needs to change the
+> way the opening fee is calculated, it can use a different field
+> in the response object, such as `opening_fee_params_v2_menu`.
+
+Clients MUST fail and abort the flow if a `opening_fee_params`
+object has unrecognized fields.
+
+`min_payment_size` and `max_payment_size` are the limits of the
+payment size, inclusive.
```

Comment:
The introduction of `payment_size` could use a note that this is the amount the sender will pay, and not the amount the client will receive. It does become clear later in the spec, but it needs some scrolling back and forth to be sure.

Reply:
Done.